
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="Documentation for CyVerse Platforms and Education">
      
      
        <meta name="author" content="CyVerse Science Team">
      
      
        <link rel="canonical" href="https://learning.cyverse.org/kallisto_tutorial/step3/">
      
      <link rel="icon" href="../assets/favicon.png">
      <meta name="generator" content="mkdocs-1.3.0, mkdocs-material-8.2.9">
    
    
      
        <title>Analyze Kallisto Results with Sleuth - Kallisto tutorial</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.120efc48.min.css">
      
        
        <link rel="stylesheet" href="../assets/stylesheets/palette.9647289d.min.css">
        
      
    
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRegular:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Regular"}</style>
      
    
    
    <script>__md_scope=new URL("..",location),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      
  


  
  


  <script>window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)},ga.l=+new Date,ga("create","UA-72224033-6","auto"),ga("set","anonymizeIp",!0),ga("send","pageview"),document.addEventListener("DOMContentLoaded",function(){document.forms.search&&document.forms.search.query.addEventListener("blur",function(){var e;this.value&&(e=document.location.pathname,ga("send","pageview",e+"?q="+this.value))}),"undefined"!=typeof location$&&location$.subscribe(function(e){ga("send","pageview",e.pathname)})})</script>
  <script async src="https://www.google-analytics.com/analytics.js"></script>


    
    
  </head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="" data-md-color-accent="">
  
    
    
      <script>var palette=__md_get("__palette");if(palette&&"object"==typeof palette.color)for(var key of Object.keys(palette.color))document.body.setAttribute("data-md-color-"+key,palette.color[key])</script>
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#analyze-kallisto-results-with-sleuth" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="Kallisto tutorial" class="md-header__button md-logo" aria-label="Kallisto tutorial" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="m20 22-3.86-1.55c.7-1.53 1.2-3.11 1.51-4.72L20 22M7.86 20.45 4 22l2.35-6.27c.31 1.61.81 3.19 1.51 4.72M12 2s5 2 5 10c0 3.1-.75 5.75-1.67 7.83A2 2 0 0 1 13.5 21h-3a2 2 0 0 1-1.83-1.17C7.76 17.75 7 15.1 7 12c0-8 5-10 5-10m0 10c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2Z"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Kallisto tutorial
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Analyze Kallisto Results with Sleuth
            
          </span>
        </div>
      </div>
    </div>
    
      <form class="md-header__option" data-md-component="palette">
        
          
          
          <input class="md-option" data-md-color-media="" data-md-color-scheme="default" data-md-color-primary="" data-md-color-accent=""  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_1">
          
            <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_2" hidden>
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M17 6H7c-3.31 0-6 2.69-6 6s2.69 6 6 6h10c3.31 0 6-2.69 6-6s-2.69-6-6-6zm0 10H7c-2.21 0-4-1.79-4-4s1.79-4 4-4h10c2.21 0 4 1.79 4 4s-1.79 4-4 4zM7 9c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"/></svg>
            </label>
          
        
          
          
          <input class="md-option" data-md-color-media="" data-md-color-scheme="slate" data-md-color-primary="" data-md-color-accent=""  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_2">
          
            <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_1" hidden>
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M17 7H7a5 5 0 0 0-5 5 5 5 0 0 0 5 5h10a5 5 0 0 0 5-5 5 5 0 0 0-5-5m0 8a3 3 0 0 1-3-3 3 3 0 0 1 3-3 3 3 0 0 1 3 3 3 3 0 0 1-3 3Z"/></svg>
            </label>
          
        
      </form>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" aria-label="Clear" tabindex="-1">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/CyVerse-learning-materials/kallisto_tutorial" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    CyVerse-learning-materials/kallisto_tutorial
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


  

<nav class="md-nav md-nav--primary md-nav--integrated" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="Kallisto tutorial" class="md-nav__button md-logo" aria-label="Kallisto tutorial" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="m20 22-3.86-1.55c.7-1.53 1.2-3.11 1.51-4.72L20 22M7.86 20.45 4 22l2.35-6.27c.31 1.61.81 3.19 1.51 4.72M12 2s5 2 5 10c0 3.1-.75 5.75-1.67 7.83A2 2 0 0 1 13.5 21h-3a2 2 0 0 1-1.83-1.17C7.76 17.75 7 15.1 7 12c0-8 5-10 5-10m0 10c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2Z"/></svg>

    </a>
    Kallisto tutorial
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/CyVerse-learning-materials/kallisto_tutorial" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    CyVerse-learning-materials/kallisto_tutorial
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href=".." class="md-nav__link">
        RNA-Seq with Kallisto and Sleuth
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../step1/" class="md-nav__link">
        Organize Kallisto Input Data
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../step2/" class="md-nav__link">
        Build Kallisto Transcriptome Index
      </a>
    </li>
  

    
      
      
      

  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" data-md-toggle="toc" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          Analyze Kallisto Results with Sleuth
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        Analyze Kallisto Results with Sleuth
      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#input-data" class="md-nav__link">
    Input  data
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#import-kallisto-results-into-r-and-analyze-with-sleuth" class="md-nav__link">
    Import Kallisto results into R and analyze with Sleuth
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
          
          <div class="md-content" data-md-component="content">
            <article class="md-content__inner md-typeset">
              
                
  <a href="https://github.com/CyVerse-learning-materials/kallisto_tutorial/edit/main/docs/step3.md" title="Edit this page" class="md-content__button md-icon">
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20.71 7.04c.39-.39.39-1.04 0-1.41l-2.34-2.34c-.37-.39-1.02-.39-1.41 0l-1.84 1.83 3.75 3.75M3 17.25V21h3.75L17.81 9.93l-3.75-3.75L3 17.25Z"/></svg>
  </a>



<h1 id="analyze-kallisto-results-with-sleuth">Analyze Kallisto Results with Sleuth<a class="headerlink" href="#analyze-kallisto-results-with-sleuth" title="Permanent link">&para;</a></h1>
<p><strong>Description:</strong></p>
<p><a href="https://pachterlab.github.io/sleuth/about">Sleuth</a> is a program for analysis of RNA-Seq experiments for which transcript abundances have been quantified with Kallisto. In this tutorial, we will
use R Studio being served from an VICE instance.</p>
<hr />
<h2 id="input-data">Input  data<a class="headerlink" href="#input-data" title="Permanent link">&para;</a></h2>
<table>
<thead>
<tr>
<th>Input</th>
<th>Description</th>
<th>Example</th>
</tr>
</thead>
<tbody>
<tr>
<td>Kallisto results folder(s)</td>
<td>Outputs from a Kallisto quantification</td>
<td><a href="http://datacommons.cyverse.org/browse/iplant/home/shared/cyverse_training/tutorials/kallisto/03_output_kallisto_results">Example directory of Kallisto results</a></td>
</tr>
</tbody>
</table>
<h2 id="import-kallisto-results-into-r-and-analyze-with-sleuth">Import Kallisto results into R and analyze with Sleuth<a class="headerlink" href="#import-kallisto-results-into-r-and-analyze-with-sleuth" title="Permanent link">&para;</a></h2>
<p>We will import the Kallisto results into an RStudio session being run
from an Atmosphere image. Then we will follow a R script based on the .</p>
<ol>
<li>If necessary, login to the CyVerse <a href="https://de.cyverse.org/de/">discovery environment</a>.</li>
<li>In the App panel, open the <strong>Sleuth RStudio</strong> app or click this
    link: <a href="https://de.cyverse.org/de/?type=quick-launch&amp;quick-launch-id=3125ee9a-9f0c-4f4c-8efd-aa6f7ea00405&amp;app-id=8eb1291c-34ea-11eb-b90c-008cfa5ae621">Sleuth app</a>.</li>
<li>Name your analysis, and if desired enter comments.</li>
<li>(Optional) In the 'Notebooks' section, under 'Select an RMarkdown
    notebook to run' select a notebook.</li>
</ol>
<div class="admonition note">
<p class="admonition-title">Sample data<p>For the sample data, navigate to and select <strong>/iplant/home/shared/cyverse_training/tutorials/kallisto/04_sleuth_R/sleuth_tutorial.Rmd</strong></p>
</p>
</div>
<p>5.  In the 'Datasets' section, under 'Data for analysis (outputs of
    Kallisto quantification)' choose the folders containing
    quantification information for all sets of reads.</p>
<div class="admonition note">
<p class="admonition-title">Sample data<p>For the sample data, navigate to and select <strong>/iplant/home/shared/cyverse_training/tutorials/kallisto/03_output_kallisto_results</strong></p>
</p>
</div>
<p>6.  In the 'Datasets' section, under 'Study design file' choose a TSV
    file describing the samples and study design (see <a href="https://pachterlab.github.io/sleuth/about">Sleuth manual</a>).</p>
<div class="admonition note">
<p class="admonition-title">Sample data<p>For the sample data, navigate to and select <strong>/iplant/home/shared/cyverse_training/tutorials/kallisto/04_sleuth_R/kallisto_demo.tsv</strong></p>
</p>
</div>
<div class="admonition tip">
<p class="admonition-title">Tip<p>See the Example study design (<a href="http://datacommons.cyverse.org/browse/iplant/home/shared/cyverse_training/tutorials/kallisto/04_sleuth_R/kallisto_demo.tsv">Kallisto demo tsv</a>) TSV file. You can create and edit
your own in a spreadsheet editing program. The <a href="https://pachterlab.github.io/sleuth/about">Sleuth manual</a>explains this file and more is described in this tutorial's RMarkdown notebook.</p>
</p>
</div>
<p>7.  If desired adjust the resources required and/or click 'Next.'</p>
<div class="admonition note">
<p class="admonition-title">Sample data<p>For the sample data, we will not specify resources.</p>
</p>
</div>
<p>8.  Click 'Launch Analyses' to start the job. Click on the Analyses
    button to monitor the job and results. In your Analyses menu, you
    will find a link to your VICE session ; this may take a few
    minutes to become active.
9.  In your RStudio session, double click on the
    <strong>sleuth_tutorial.Rmd</strong> file and follow the tutorial by pressing
    the green "play" triangles in each section of code. The code for
    the notebook is replicated below:</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-0-1" name="__codelineno-0-1" href="#__codelineno-0-1"></a>---
<a id="__codelineno-0-2" name="__codelineno-0-2" href="#__codelineno-0-2"></a>title: &quot;Sleuth RNA-Seq Tutorial - Arabidopsis&quot; 
<a id="__codelineno-0-3" name="__codelineno-0-3" href="#__codelineno-0-3"></a>output:
<a id="__codelineno-0-4" name="__codelineno-0-4" href="#__codelineno-0-4"></a>    html_document: 
<a id="__codelineno-0-5" name="__codelineno-0-5" href="#__codelineno-0-5"></a>        df_print: paged 
<a id="__codelineno-0-6" name="__codelineno-0-6" href="#__codelineno-0-6"></a>---
<a id="__codelineno-0-7" name="__codelineno-0-7" href="#__codelineno-0-7"></a>
<a id="__codelineno-0-8" name="__codelineno-0-8" href="#__codelineno-0-8"></a>This tutorial will take you through a sample RNA-Seq analysis
<a id="__codelineno-0-9" name="__codelineno-0-9" href="#__codelineno-0-9"></a>using [kallisto](&lt;https://pachterlab.github.io/kallisto/about&gt;),
<a id="__codelineno-0-10" name="__codelineno-0-10" href="#__codelineno-0-10"></a>using an RNA-Seq R package [Sleuth](&lt;https://pachterlab.github.io/sleuth/about&gt;). 
<a id="__codelineno-0-11" name="__codelineno-0-11" href="#__codelineno-0-11"></a>This tutorial is based on the one by the [Pachterlab](&lt;https://pachterlab.github.io/sleuth_walkthroughs/boj/analysis.html&gt;)
<a id="__codelineno-0-12" name="__codelineno-0-12" href="#__codelineno-0-12"></a>
<a id="__codelineno-0-13" name="__codelineno-0-13" href="#__codelineno-0-13"></a>## Step 1: Load Sleuth and accessory libraries
<a id="__codelineno-0-14" name="__codelineno-0-14" href="#__codelineno-0-14"></a>
<a id="__codelineno-0-15" name="__codelineno-0-15" href="#__codelineno-0-15"></a>Next, we need to load the Sleuth library to begin. We will also
<a id="__codelineno-0-16" name="__codelineno-0-16" href="#__codelineno-0-16"></a>check the version:
<a id="__codelineno-0-17" name="__codelineno-0-17" href="#__codelineno-0-17"></a>
<a id="__codelineno-0-18" name="__codelineno-0-18" href="#__codelineno-0-18"></a>```{r message=FALSE} 
<a id="__codelineno-0-19" name="__codelineno-0-19" href="#__codelineno-0-19"></a>require(&quot;sleuth&quot;) 
<a id="__codelineno-0-20" name="__codelineno-0-20" href="#__codelineno-0-20"></a>packageVersion(&quot;sleuth&quot;) 
<a id="__codelineno-0-21" name="__codelineno-0-21" href="#__codelineno-0-21"></a>```
<a id="__codelineno-0-22" name="__codelineno-0-22" href="#__codelineno-0-22"></a>
<a id="__codelineno-0-23" name="__codelineno-0-23" href="#__codelineno-0-23"></a>We will also install a plotting library and some other functions
<a id="__codelineno-0-24" name="__codelineno-0-24" href="#__codelineno-0-24"></a>we will need...
<a id="__codelineno-0-25" name="__codelineno-0-25" href="#__codelineno-0-25"></a>
<a id="__codelineno-0-26" name="__codelineno-0-26" href="#__codelineno-0-26"></a>```{r echo=FALSE, message=FALSE, warning=FALSE} 
<a id="__codelineno-0-27" name="__codelineno-0-27" href="#__codelineno-0-27"></a>library(&quot;gridExtra&quot;) 
<a id="__codelineno-0-28" name="__codelineno-0-28" href="#__codelineno-0-28"></a>library(&quot;cowplot&quot;) 
<a id="__codelineno-0-29" name="__codelineno-0-29" href="#__codelineno-0-29"></a>```
<a id="__codelineno-0-30" name="__codelineno-0-30" href="#__codelineno-0-30"></a>
<a id="__codelineno-0-31" name="__codelineno-0-31" href="#__codelineno-0-31"></a>We will also use
<a id="__codelineno-0-32" name="__codelineno-0-32" href="#__codelineno-0-32"></a>[biomaRt](&lt;https://bioconductor.org/packages/release/bioc/html/biomaRt.html&gt;)
<a id="__codelineno-0-33" name="__codelineno-0-33" href="#__codelineno-0-33"></a>tools will allow us to pull in recognizable gene names from a
<a id="__codelineno-0-34" name="__codelineno-0-34" href="#__codelineno-0-34"></a>database.
<a id="__codelineno-0-35" name="__codelineno-0-35" href="#__codelineno-0-35"></a>
<a id="__codelineno-0-36" name="__codelineno-0-36" href="#__codelineno-0-36"></a>```{r echo=FALSE, message=FALSE, warning=FALSE} 
<a id="__codelineno-0-37" name="__codelineno-0-37" href="#__codelineno-0-37"></a>library(&quot;biomaRt&quot;) 
<a id="__codelineno-0-38" name="__codelineno-0-38" href="#__codelineno-0-38"></a>```
<a id="__codelineno-0-39" name="__codelineno-0-39" href="#__codelineno-0-39"></a>
<a id="__codelineno-0-40" name="__codelineno-0-40" href="#__codelineno-0-40"></a>## Step 2: Load experimental design and label kallisto outputs
<a id="__codelineno-0-41" name="__codelineno-0-41" href="#__codelineno-0-41"></a>with metadata
<a id="__codelineno-0-42" name="__codelineno-0-42" href="#__codelineno-0-42"></a>
<a id="__codelineno-0-43" name="__codelineno-0-43" href="#__codelineno-0-43"></a>### Locate sample names and describe our experimental design
<a id="__codelineno-0-44" name="__codelineno-0-44" href="#__codelineno-0-44"></a>
<a id="__codelineno-0-45" name="__codelineno-0-45" href="#__codelineno-0-45"></a>We need to provide Sleuth with our sample names:
<a id="__codelineno-0-46" name="__codelineno-0-46" href="#__codelineno-0-46"></a>
<a id="__codelineno-0-47" name="__codelineno-0-47" href="#__codelineno-0-47"></a>```{r} 
<a id="__codelineno-0-48" name="__codelineno-0-48" href="#__codelineno-0-48"></a>sample_id &lt;- dir(file.path(&quot;~/kallisto_qaunt_output/&quot;)) 
<a id="__codelineno-0-49" name="__codelineno-0-49" href="#__codelineno-0-49"></a>sample_id 
<a id="__codelineno-0-50" name="__codelineno-0-50" href="#__codelineno-0-50"></a>```
<a id="__codelineno-0-51" name="__codelineno-0-51" href="#__codelineno-0-51"></a>
<a id="__codelineno-0-52" name="__codelineno-0-52" href="#__codelineno-0-52"></a>We also need to get the file paths to our results files.
<a id="__codelineno-0-53" name="__codelineno-0-53" href="#__codelineno-0-53"></a>
<a id="__codelineno-0-54" name="__codelineno-0-54" href="#__codelineno-0-54"></a>```{r} 
<a id="__codelineno-0-55" name="__codelineno-0-55" href="#__codelineno-0-55"></a>kal_dirs &lt;- file.path(&quot;~/kallisto_qaunt_output&quot;, sample_id) 
<a id="__codelineno-0-56" name="__codelineno-0-56" href="#__codelineno-0-56"></a>```
<a id="__codelineno-0-57" name="__codelineno-0-57" href="#__codelineno-0-57"></a>
<a id="__codelineno-0-58" name="__codelineno-0-58" href="#__codelineno-0-58"></a>We also need a table that provides more meaningful names for
<a id="__codelineno-0-59" name="__codelineno-0-59" href="#__codelineno-0-59"></a>describing our experiment...
<a id="__codelineno-0-60" name="__codelineno-0-60" href="#__codelineno-0-60"></a>
<a id="__codelineno-0-61" name="__codelineno-0-61" href="#__codelineno-0-61"></a>```{r} 
<a id="__codelineno-0-62" name="__codelineno-0-62" href="#__codelineno-0-62"></a>s2c &lt;- read.table(file.path(&quot;~/kallisto_demo.tsv&quot;),
<a id="__codelineno-0-63" name="__codelineno-0-63" href="#__codelineno-0-63"></a>                            header = TRUE,
<a id="__codelineno-0-64" name="__codelineno-0-64" href="#__codelineno-0-64"></a>                            stringsAsFactors = FALSE,
<a id="__codelineno-0-65" name="__codelineno-0-65" href="#__codelineno-0-65"></a>                            sep = &quot;\t&quot;) 
<a id="__codelineno-0-66" name="__codelineno-0-66" href="#__codelineno-0-66"></a>```
<a id="__codelineno-0-67" name="__codelineno-0-67" href="#__codelineno-0-67"></a>
<a id="__codelineno-0-68" name="__codelineno-0-68" href="#__codelineno-0-68"></a>We will add our file paths to the table
<a id="__codelineno-0-69" name="__codelineno-0-69" href="#__codelineno-0-69"></a>
<a id="__codelineno-0-70" name="__codelineno-0-70" href="#__codelineno-0-70"></a>```{r} 
<a id="__codelineno-0-71" name="__codelineno-0-71" href="#__codelineno-0-71"></a>s2c &lt;- dplyr::mutate(s2c, path = kal_dirs) 
<a id="__codelineno-0-72" name="__codelineno-0-72" href="#__codelineno-0-72"></a>```
<a id="__codelineno-0-73" name="__codelineno-0-73" href="#__codelineno-0-73"></a>
<a id="__codelineno-0-74" name="__codelineno-0-74" href="#__codelineno-0-74"></a>Let&#39;s view the table we have created: 
<a id="__codelineno-0-75" name="__codelineno-0-75" href="#__codelineno-0-75"></a>
<a id="__codelineno-0-76" name="__codelineno-0-76" href="#__codelineno-0-76"></a>```{r} 
<a id="__codelineno-0-77" name="__codelineno-0-77" href="#__codelineno-0-77"></a>s2c 
<a id="__codelineno-0-78" name="__codelineno-0-78" href="#__codelineno-0-78"></a>```
<a id="__codelineno-0-79" name="__codelineno-0-79" href="#__codelineno-0-79"></a>## Step 3: Load gene names from Ensembl
<a id="__codelineno-0-80" name="__codelineno-0-80" href="#__codelineno-0-80"></a>
<a id="__codelineno-0-81" name="__codelineno-0-81" href="#__codelineno-0-81"></a>Next we need to determine which biomaRt to use. This can be a
<a id="__codelineno-0-82" name="__codelineno-0-82" href="#__codelineno-0-82"></a>little complex so be sure to read their
<a id="__codelineno-0-83" name="__codelineno-0-83" href="#__codelineno-0-83"></a>[documentation](&lt;https://www.bioconductor.org/packages/devel/bioc/vignettes/biomaRt/inst/doc/biomaRt.html&gt;).
<a id="__codelineno-0-84" name="__codelineno-0-84" href="#__codelineno-0-84"></a>This [blog post](&lt;https://nsaunders.wordpress.com/2015/04/28/some-basics-of-biomart/&gt;)
<a id="__codelineno-0-85" name="__codelineno-0-85" href="#__codelineno-0-85"></a>is also helpful.
<a id="__codelineno-0-86" name="__codelineno-0-86" href="#__codelineno-0-86"></a>
<a id="__codelineno-0-87" name="__codelineno-0-87" href="#__codelineno-0-87"></a>```{r} 
<a id="__codelineno-0-88" name="__codelineno-0-88" href="#__codelineno-0-88"></a>marts &lt;- listMarts() 
<a id="__codelineno-0-89" name="__codelineno-0-89" href="#__codelineno-0-89"></a>marts 
<a id="__codelineno-0-90" name="__codelineno-0-90" href="#__codelineno-0-90"></a>```
<a id="__codelineno-0-91" name="__codelineno-0-91" href="#__codelineno-0-91"></a>
<a id="__codelineno-0-92" name="__codelineno-0-92" href="#__codelineno-0-92"></a>If you are not working with these Ensembl data bases you may want
<a id="__codelineno-0-93" name="__codelineno-0-93" href="#__codelineno-0-93"></a>to check out documentation on [using BiomaRts other than
<a id="__codelineno-0-94" name="__codelineno-0-94" href="#__codelineno-0-94"></a>Ensembl](&lt;https://bioconductor.org/packages/release/bioc/vignettes/biomaRt/inst/doc/biomaRt.html#using-a-biomart-other-than-ensembl&gt;).
<a id="__codelineno-0-95" name="__codelineno-0-95" href="#__codelineno-0-95"></a>We are using plants, so...
<a id="__codelineno-0-96" name="__codelineno-0-96" href="#__codelineno-0-96"></a>
<a id="__codelineno-0-97" name="__codelineno-0-97" href="#__codelineno-0-97"></a>```{r} 
<a id="__codelineno-0-98" name="__codelineno-0-98" href="#__codelineno-0-98"></a>marts &lt;- listMarts(host = &quot;plants.ensembl.org&quot;) 
<a id="__codelineno-0-99" name="__codelineno-0-99" href="#__codelineno-0-99"></a>marts 
<a id="__codelineno-0-100" name="__codelineno-0-100" href="#__codelineno-0-100"></a>```
<a id="__codelineno-0-101" name="__codelineno-0-101" href="#__codelineno-0-101"></a>
<a id="__codelineno-0-102" name="__codelineno-0-102" href="#__codelineno-0-102"></a>For now, remember that we will want to use plants_mart.
<a id="__codelineno-0-103" name="__codelineno-0-103" href="#__codelineno-0-103"></a>
<a id="__codelineno-0-104" name="__codelineno-0-104" href="#__codelineno-0-104"></a>Next, we need to choose a specific data set.
<a id="__codelineno-0-105" name="__codelineno-0-105" href="#__codelineno-0-105"></a>
<a id="__codelineno-0-106" name="__codelineno-0-106" href="#__codelineno-0-106"></a>```{r}
<a id="__codelineno-0-107" name="__codelineno-0-107" href="#__codelineno-0-107"></a>plants_mart &lt;- useMart(&quot;plants_mart&quot;, host = &quot;plants.ensembl.org&quot; ) 
<a id="__codelineno-0-108" name="__codelineno-0-108" href="#__codelineno-0-108"></a>listDatasets(plants_mart) 
<a id="__codelineno-0-109" name="__codelineno-0-109" href="#__codelineno-0-109"></a>```
<a id="__codelineno-0-110" name="__codelineno-0-110" href="#__codelineno-0-110"></a>
<a id="__codelineno-0-111" name="__codelineno-0-111" href="#__codelineno-0-111"></a>After a little looking, its the athaliana_eg_gene data set that
<a id="__codelineno-0-112" name="__codelineno-0-112" href="#__codelineno-0-112"></a>we need. Finally, we need to update our plants_mart to be more
<a id="__codelineno-0-113" name="__codelineno-0-113" href="#__codelineno-0-113"></a>specific.
<a id="__codelineno-0-114" name="__codelineno-0-114" href="#__codelineno-0-114"></a>
<a id="__codelineno-0-115" name="__codelineno-0-115" href="#__codelineno-0-115"></a>```{r} 
<a id="__codelineno-0-116" name="__codelineno-0-116" href="#__codelineno-0-116"></a>plants_mart &lt;- useMart(&quot;plants_mart&quot;, dataset = &quot;athaliana_eg_gene&quot;, host=&quot;plants.ensembl.org&quot; ) 
<a id="__codelineno-0-117" name="__codelineno-0-117" href="#__codelineno-0-117"></a>```
<a id="__codelineno-0-118" name="__codelineno-0-118" href="#__codelineno-0-118"></a>
<a id="__codelineno-0-119" name="__codelineno-0-119" href="#__codelineno-0-119"></a>Now we want to get specific attributes from the list of genes we
<a id="__codelineno-0-120" name="__codelineno-0-120" href="#__codelineno-0-120"></a>can import from biomart
<a id="__codelineno-0-121" name="__codelineno-0-121" href="#__codelineno-0-121"></a>
<a id="__codelineno-0-122" name="__codelineno-0-122" href="#__codelineno-0-122"></a>```{r} 
<a id="__codelineno-0-123" name="__codelineno-0-123" href="#__codelineno-0-123"></a>listAttributes(plants_mart) 
<a id="__codelineno-0-124" name="__codelineno-0-124" href="#__codelineno-0-124"></a>```
<a id="__codelineno-0-125" name="__codelineno-0-125" href="#__codelineno-0-125"></a>
<a id="__codelineno-0-126" name="__codelineno-0-126" href="#__codelineno-0-126"></a>We can choose whichever of these we&#39;d like to use. Let&#39;s get
<a id="__codelineno-0-127" name="__codelineno-0-127" href="#__codelineno-0-127"></a>transcript ids, gene ids, a description, and gene names. Notice,
<a id="__codelineno-0-128" name="__codelineno-0-128" href="#__codelineno-0-128"></a>there are many things you may want to come back for. We must get
<a id="__codelineno-0-129" name="__codelineno-0-129" href="#__codelineno-0-129"></a>the transcript id because these are the names of the transcripts
<a id="__codelineno-0-130" name="__codelineno-0-130" href="#__codelineno-0-130"></a>that were used in our Kallisto quantification.
<a id="__codelineno-0-131" name="__codelineno-0-131" href="#__codelineno-0-131"></a>
<a id="__codelineno-0-132" name="__codelineno-0-132" href="#__codelineno-0-132"></a>```{r echo=FALSE, message=FALSE, warning=FALSE} 
<a id="__codelineno-0-133" name="__codelineno-0-133" href="#__codelineno-0-133"></a>t2g &lt;- getBM(attributes = c(&quot;ensembl_transcript_id&quot;,                                            
<a id="__codelineno-0-134" name="__codelineno-0-134" href="#__codelineno-0-134"></a>                            &quot;ensembl_gene_id&quot;,                             
<a id="__codelineno-0-135" name="__codelineno-0-135" href="#__codelineno-0-135"></a>                            &quot;description&quot;,
<a id="__codelineno-0-136" name="__codelineno-0-136" href="#__codelineno-0-136"></a>                            &quot;external_gene_name&quot;),              
<a id="__codelineno-0-137" name="__codelineno-0-137" href="#__codelineno-0-137"></a>                            mart = plants_mart)
<a id="__codelineno-0-138" name="__codelineno-0-138" href="#__codelineno-0-138"></a>```
<a id="__codelineno-0-139" name="__codelineno-0-139" href="#__codelineno-0-139"></a>
<a id="__codelineno-0-140" name="__codelineno-0-140" href="#__codelineno-0-140"></a>We need to make sure the ensembl_transcript_id column is named
<a id="__codelineno-0-141" name="__codelineno-0-141" href="#__codelineno-0-141"></a>target_id
<a id="__codelineno-0-142" name="__codelineno-0-142" href="#__codelineno-0-142"></a>
<a id="__codelineno-0-143" name="__codelineno-0-143" href="#__codelineno-0-143"></a>```{r} 
<a id="__codelineno-0-144" name="__codelineno-0-144" href="#__codelineno-0-144"></a>ttg &lt;- dplyr::rename(t2g, target_id= ensembl_transcript_id, ens_gene = ensembl_gene_id, ext_gene = external_gene_name) 
<a id="__codelineno-0-145" name="__codelineno-0-145" href="#__codelineno-0-145"></a>```
<a id="__codelineno-0-146" name="__codelineno-0-146" href="#__codelineno-0-146"></a>
<a id="__codelineno-0-147" name="__codelineno-0-147" href="#__codelineno-0-147"></a>##Step 4: Prepare data for Sleuth
<a id="__codelineno-0-148" name="__codelineno-0-148" href="#__codelineno-0-148"></a>
<a id="__codelineno-0-149" name="__codelineno-0-149" href="#__codelineno-0-149"></a>first we need to alter our experimental design so that we consider
<a id="__codelineno-0-150" name="__codelineno-0-150" href="#__codelineno-0-150"></a>the full transcriptome sample to be the &quot;control&quot; to compare to...
<a id="__codelineno-0-151" name="__codelineno-0-151" href="#__codelineno-0-151"></a>
<a id="__codelineno-0-152" name="__codelineno-0-152" href="#__codelineno-0-152"></a>```{r} 
<a id="__codelineno-0-153" name="__codelineno-0-153" href="#__codelineno-0-153"></a>s2c$genotype_variation_s &lt;- as.factor(s2c$genotype_variation_s) 
<a id="__codelineno-0-154" name="__codelineno-0-154" href="#__codelineno-0-154"></a>s2c$genotype_variation_s &lt;- relevel(s2c$genotype_variation_s, ref = &quot;wild type&quot;) 
<a id="__codelineno-0-155" name="__codelineno-0-155" href="#__codelineno-0-155"></a>```
<a id="__codelineno-0-156" name="__codelineno-0-156" href="#__codelineno-0-156"></a>
<a id="__codelineno-0-157" name="__codelineno-0-157" href="#__codelineno-0-157"></a>Now we need to tell Sleuth both about the Kallisto results and the
<a id="__codelineno-0-158" name="__codelineno-0-158" href="#__codelineno-0-158"></a>gene names (and gene descriptions/metadata) we obtained from
<a id="__codelineno-0-159" name="__codelineno-0-159" href="#__codelineno-0-159"></a>biomaRt. The sleuth_prep function does this.
<a id="__codelineno-0-160" name="__codelineno-0-160" href="#__codelineno-0-160"></a>
<a id="__codelineno-0-161" name="__codelineno-0-161" href="#__codelineno-0-161"></a>```{r warning=FALSE} so &lt;- sleuth_prep(s2c,
<a id="__codelineno-0-162" name="__codelineno-0-162" href="#__codelineno-0-162"></a>                                        full_model = ~genotype_variation_s,
<a id="__codelineno-0-163" name="__codelineno-0-163" href="#__codelineno-0-163"></a>                                        target_mapping = ttg,
<a id="__codelineno-0-164" name="__codelineno-0-164" href="#__codelineno-0-164"></a>                                        read_bootstrap_tpm=TRUE,  
<a id="__codelineno-0-165" name="__codelineno-0-165" href="#__codelineno-0-165"></a>                                        extra_bootstrap_summary = TRUE)
<a id="__codelineno-0-166" name="__codelineno-0-166" href="#__codelineno-0-166"></a>```
<a id="__codelineno-0-167" name="__codelineno-0-167" href="#__codelineno-0-167"></a>
<a id="__codelineno-0-168" name="__codelineno-0-168" href="#__codelineno-0-168"></a>##Step 5: Initial data exploration
<a id="__codelineno-0-169" name="__codelineno-0-169" href="#__codelineno-0-169"></a>
<a id="__codelineno-0-170" name="__codelineno-0-170" href="#__codelineno-0-170"></a>### Examine Sleuth PCA
<a id="__codelineno-0-171" name="__codelineno-0-171" href="#__codelineno-0-171"></a>
<a id="__codelineno-0-172" name="__codelineno-0-172" href="#__codelineno-0-172"></a>Next, we should check to see if our samples (and replicates)
<a id="__codelineno-0-173" name="__codelineno-0-173" href="#__codelineno-0-173"></a>cluster on a PCA (as should expect) or if there are outliers. When
<a id="__codelineno-0-174" name="__codelineno-0-174" href="#__codelineno-0-174"></a>we plot by condition, we&#39;d expect that similar colors group
<a id="__codelineno-0-175" name="__codelineno-0-175" href="#__codelineno-0-175"></a>together.
<a id="__codelineno-0-176" name="__codelineno-0-176" href="#__codelineno-0-176"></a>
<a id="__codelineno-0-177" name="__codelineno-0-177" href="#__codelineno-0-177"></a>```{r} 
<a id="__codelineno-0-178" name="__codelineno-0-178" href="#__codelineno-0-178"></a>library(cowplot) 
<a id="__codelineno-0-179" name="__codelineno-0-179" href="#__codelineno-0-179"></a>ggplot2::theme_set(theme_cowplot()) 
<a id="__codelineno-0-180" name="__codelineno-0-180" href="#__codelineno-0-180"></a>plot_pca(so, color_by = &#39;genotype_variation_s&#39;, text_labels = TRUE) 
<a id="__codelineno-0-181" name="__codelineno-0-181" href="#__codelineno-0-181"></a>```
<a id="__codelineno-0-182" name="__codelineno-0-182" href="#__codelineno-0-182"></a>
<a id="__codelineno-0-183" name="__codelineno-0-183" href="#__codelineno-0-183"></a>Let&#39;s try plotting by treatment
<a id="__codelineno-0-184" name="__codelineno-0-184" href="#__codelineno-0-184"></a>
<a id="__codelineno-0-185" name="__codelineno-0-185" href="#__codelineno-0-185"></a>```{r} 
<a id="__codelineno-0-186" name="__codelineno-0-186" href="#__codelineno-0-186"></a>plot_pca(so, color_by = &#39;treatment_s&#39;, text_labels = TRUE) 
<a id="__codelineno-0-187" name="__codelineno-0-187" href="#__codelineno-0-187"></a>```
<a id="__codelineno-0-188" name="__codelineno-0-188" href="#__codelineno-0-188"></a>
<a id="__codelineno-0-189" name="__codelineno-0-189" href="#__codelineno-0-189"></a>We can also see genes involved in the the 1st PC by looking at the
<a id="__codelineno-0-190" name="__codelineno-0-190" href="#__codelineno-0-190"></a>loadings (primary genes whose linear combinations define the
<a id="__codelineno-0-191" name="__codelineno-0-191" href="#__codelineno-0-191"></a>principal components)
<a id="__codelineno-0-192" name="__codelineno-0-192" href="#__codelineno-0-192"></a>
<a id="__codelineno-0-193" name="__codelineno-0-193" href="#__codelineno-0-193"></a>```{r} 
<a id="__codelineno-0-194" name="__codelineno-0-194" href="#__codelineno-0-194"></a>plot_loadings(so, pc_input = 1) 
<a id="__codelineno-0-195" name="__codelineno-0-195" href="#__codelineno-0-195"></a>```
<a id="__codelineno-0-196" name="__codelineno-0-196" href="#__codelineno-0-196"></a>
<a id="__codelineno-0-197" name="__codelineno-0-197" href="#__codelineno-0-197"></a>Let&#39;s see how this &quot;influential&quot; gene (at least as far as PCA
<a id="__codelineno-0-198" name="__codelineno-0-198" href="#__codelineno-0-198"></a>tells us) looks by condition
<a id="__codelineno-0-199" name="__codelineno-0-199" href="#__codelineno-0-199"></a>
<a id="__codelineno-0-200" name="__codelineno-0-200" href="#__codelineno-0-200"></a>```{r} 
<a id="__codelineno-0-201" name="__codelineno-0-201" href="#__codelineno-0-201"></a>plot_bootstrap(so, &#39;AT2G34420.1&#39;, color_by = &#39;genotype_variation_s&#39;) 
<a id="__codelineno-0-202" name="__codelineno-0-202" href="#__codelineno-0-202"></a>```
<a id="__codelineno-0-203" name="__codelineno-0-203" href="#__codelineno-0-203"></a>
<a id="__codelineno-0-204" name="__codelineno-0-204" href="#__codelineno-0-204"></a>Let&#39;s see how this &quot;influential&quot; gene (at least as far as PCA
<a id="__codelineno-0-205" name="__codelineno-0-205" href="#__codelineno-0-205"></a>tells us) looks by treatment
<a id="__codelineno-0-206" name="__codelineno-0-206" href="#__codelineno-0-206"></a>
<a id="__codelineno-0-207" name="__codelineno-0-207" href="#__codelineno-0-207"></a>```{r} 
<a id="__codelineno-0-208" name="__codelineno-0-208" href="#__codelineno-0-208"></a>plot_bootstrap(so, &#39;AT2G34420.1&#39;, color_by = &#39;treatment_s&#39;) 
<a id="__codelineno-0-209" name="__codelineno-0-209" href="#__codelineno-0-209"></a>```
<a id="__codelineno-0-210" name="__codelineno-0-210" href="#__codelineno-0-210"></a>
<a id="__codelineno-0-211" name="__codelineno-0-211" href="#__codelineno-0-211"></a>##Step 6: Modeling, testing, and results exploration
<a id="__codelineno-0-212" name="__codelineno-0-212" href="#__codelineno-0-212"></a>
<a id="__codelineno-0-213" name="__codelineno-0-213" href="#__codelineno-0-213"></a>### Differential expression testing with Sleuth
<a id="__codelineno-0-214" name="__codelineno-0-214" href="#__codelineno-0-214"></a>
<a id="__codelineno-0-215" name="__codelineno-0-215" href="#__codelineno-0-215"></a>Now we need to run a few functions that will test for differential
<a id="__codelineno-0-216" name="__codelineno-0-216" href="#__codelineno-0-216"></a>expression (abundance).
<a id="__codelineno-0-217" name="__codelineno-0-217" href="#__codelineno-0-217"></a>
<a id="__codelineno-0-218" name="__codelineno-0-218" href="#__codelineno-0-218"></a>First we will create a model
<a id="__codelineno-0-219" name="__codelineno-0-219" href="#__codelineno-0-219"></a>
<a id="__codelineno-0-220" name="__codelineno-0-220" href="#__codelineno-0-220"></a>```{r} 
<a id="__codelineno-0-221" name="__codelineno-0-221" href="#__codelineno-0-221"></a>so &lt;- sleuth_fit(so, ~genotype_variation_s, &#39;full&#39;) 
<a id="__codelineno-0-222" name="__codelineno-0-222" href="#__codelineno-0-222"></a>so &lt;- sleuth_fit(so, ~1, &#39;reduced&#39;) 
<a id="__codelineno-0-223" name="__codelineno-0-223" href="#__codelineno-0-223"></a>so &lt;- sleuth_lrt(so, &#39;reduced&#39;, &#39;full&#39;) 
<a id="__codelineno-0-224" name="__codelineno-0-224" href="#__codelineno-0-224"></a>```
<a id="__codelineno-0-225" name="__codelineno-0-225" href="#__codelineno-0-225"></a>
<a id="__codelineno-0-226" name="__codelineno-0-226" href="#__codelineno-0-226"></a>Now we can get the results of this analysis
<a id="__codelineno-0-227" name="__codelineno-0-227" href="#__codelineno-0-227"></a>
<a id="__codelineno-0-228" name="__codelineno-0-228" href="#__codelineno-0-228"></a>```{r} 
<a id="__codelineno-0-229" name="__codelineno-0-229" href="#__codelineno-0-229"></a>full_results &lt;- sleuth_results(so, &#39;reduced:full&#39;, &#39;lrt&#39;,
<a id="__codelineno-0-230" name="__codelineno-0-230" href="#__codelineno-0-230"></a>                            show_all = FALSE) 
<a id="__codelineno-0-231" name="__codelineno-0-231" href="#__codelineno-0-231"></a>head(full_results) 
<a id="__codelineno-0-232" name="__codelineno-0-232" href="#__codelineno-0-232"></a>```
<a id="__codelineno-0-233" name="__codelineno-0-233" href="#__codelineno-0-233"></a>
<a id="__codelineno-0-234" name="__codelineno-0-234" href="#__codelineno-0-234"></a>Let&#39;s add Wald test
<a id="__codelineno-0-235" name="__codelineno-0-235" href="#__codelineno-0-235"></a>```{r} 
<a id="__codelineno-0-236" name="__codelineno-0-236" href="#__codelineno-0-236"></a>wald_test &lt;- colnames(design_matrix(so))[2]
<a id="__codelineno-0-237" name="__codelineno-0-237" href="#__codelineno-0-237"></a>so &lt;- sleuth_wt(so, wald_test)
<a id="__codelineno-0-238" name="__codelineno-0-238" href="#__codelineno-0-238"></a>```
<a id="__codelineno-0-239" name="__codelineno-0-239" href="#__codelineno-0-239"></a>
<a id="__codelineno-0-240" name="__codelineno-0-240" href="#__codelineno-0-240"></a>And start a Shiny Browser
<a id="__codelineno-0-241" name="__codelineno-0-241" href="#__codelineno-0-241"></a>
<a id="__codelineno-0-242" name="__codelineno-0-242" href="#__codelineno-0-242"></a>```{r} 
<a id="__codelineno-0-243" name="__codelineno-0-243" href="#__codelineno-0-243"></a>sleuth_live(so) 
<a id="__codelineno-0-244" name="__codelineno-0-244" href="#__codelineno-0-244"></a>```
</code></pre></div>
<hr />
<p><strong>Summary</strong> </p>
<p>Together, Kallisto and Sleuth are quick, powerful ways to
analyze RNA-Seq data.</p>

              
            </article>
          </div>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
    <nav class="md-footer__inner md-grid" aria-label="Footer">
      
        
        <a href="../step2/" class="md-footer__link md-footer__link--prev" aria-label="Previous: Build Kallisto Transcriptome Index" rel="prev">
          <div class="md-footer__button md-icon">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
          </div>
          <div class="md-footer__title">
            <div class="md-ellipsis">
              <span class="md-footer__direction">
                Previous
              </span>
              Build Kallisto Transcriptome Index
            </div>
          </div>
        </a>
      
      
    </nav>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      Copyright &copy; 2022 CyVerse
    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
        <div class="md-social">
  
    
    
      
      
    
    <a href="https://orcid.org/" target="_blank" rel="noopener" title="orcid.org" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M294.75 188.19h-45.92V342h47.47c67.62 0 83.12-51.34 83.12-76.91 0-41.64-26.54-76.9-84.67-76.9zM256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm-80.79 360.76h-29.84v-207.5h29.84zm-14.92-231.14a19.57 19.57 0 1 1 19.57-19.57 19.64 19.64 0 0 1-19.57 19.57zM300 369h-81V161.26h80.6c76.73 0 110.44 54.83 110.44 103.85C410 318.39 368.38 369 300 369z"/></svg>
    </a>
  
    
    
      
      
    
    <a href="https://github.com/cyverse-learning-material/" target="_blank" rel="noopener" title="github.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><!--! Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg>
    </a>
  
    
    
      
      
    
    <a href="https://twitter.com/cyverseorg" target="_blank" rel="noopener" title="twitter.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"/></svg>
    </a>
  
    
    
      
      
    
    <a href="https://www.linkedin.com/company/cyverse.org" target="_blank" rel="noopener" title="www.linkedin.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9V416z"/></svg>
    </a>
  
    
    
      
      
    
    <a href="https://harbor.cyverse.org/" target="_blank" rel="noopener" title="harbor.cyverse.org" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 640 512"><!--! Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M349.9 236.3h-66.1v-59.4h66.1v59.4zm0-204.3h-66.1v60.7h66.1V32zm78.2 144.8H362v59.4h66.1v-59.4zm-156.3-72.1h-66.1v60.1h66.1v-60.1zm78.1 0h-66.1v60.1h66.1v-60.1zm276.8 100c-14.4-9.7-47.6-13.2-73.1-8.4-3.3-24-16.7-44.9-41.1-63.7l-14-9.3-9.3 14c-18.4 27.8-23.4 73.6-3.7 103.8-8.7 4.7-25.8 11.1-48.4 10.7H2.4c-8.7 50.8 5.8 116.8 44 162.1 37.1 43.9 92.7 66.2 165.4 66.2 157.4 0 273.9-72.5 328.4-204.2 21.4.4 67.6.1 91.3-45.2 1.5-2.5 6.6-13.2 8.5-17.1l-13.3-8.9zm-511.1-27.9h-66v59.4h66.1v-59.4zm78.1 0h-66.1v59.4h66.1v-59.4zm78.1 0h-66.1v59.4h66.1v-59.4zm-78.1-72.1h-66.1v60.1h66.1v-60.1z"/></svg>
    </a>
  
    
    
      
      
    
    <a href="https://www.youtube.com/c/CyverseOrgProject" target="_blank" rel="noopener" title="www.youtube.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 576 512"><!--! Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M549.655 124.083c-6.281-23.65-24.787-42.276-48.284-48.597C458.781 64 288 64 288 64S117.22 64 74.629 75.486c-23.497 6.322-42.003 24.947-48.284 48.597-11.412 42.867-11.412 132.305-11.412 132.305s0 89.438 11.412 132.305c6.281 23.65 24.787 41.5 48.284 47.821C117.22 448 288 448 288 448s170.78 0 213.371-11.486c23.497-6.321 42.003-24.171 48.284-47.821 11.412-42.867 11.412-132.305 11.412-132.305s0-89.438-11.412-132.305zm-317.51 213.508V175.185l142.739 81.205-142.739 81.201z"/></svg>
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    <script id="__config" type="application/json">{"base": "..", "features": ["navigation.instant", "navigation.tracking", "navigation.indexes", "toc.integrate", "toc.follow"], "search": "../assets/javascripts/workers/search.2a1c317c.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.placeholder": "Search", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version.title": "Select version"}}</script>
    
    
      <script src="../assets/javascripts/bundle.6e54b5cd.min.js"></script>
      
        <script src="../javascripts/intercom-learning.js"></script>
      
    
  </body>
</html>